### PR TITLE
update otel go dependency to v1.0.0-RC1

### PR DESCRIPTION
**Pull request by codeboten**
Bumping the depency of OpenTelemetry go to v1.0.0 release candidate 1 in preparation for the 1.0.0 release.

### CODE REVIEW

**Review by ahayworth — APPROVED** [review]
Seems alright to me. Since we don't (yet?!) have any tests; I presume you have tested this locally and that it works?

**Comment by codeboten** [comment]
Yep, ran the command manually and it all worked as it did before.

**Comment by tobert** [comment]
There are unit tests in cmd/ and some functional tests in the `demos` directory, if all those play out, we're probably good. I'll make an issue to start shaping those demos into actual tests we can automate.

File: cmd/plumbing.go [hunk 1]
```diff
@@ -28,51 +26,49 @@ import (
...
+		grpcOpts = append(grpcOpts, otlpgrpc.WithHeaders(otlpHeaders))
 	}
 
 	// OTLP examples usually show this with the grpc.WithBlock() dial option to
 	// make the connection synchronous, but it's not the right default for cli
 	// instead, rely on the shutdown methods to make sure everything is flushed
 	// by the time the program exits.
 	if otlpBlocking {
-		driverOpts = append(driverOpts, otlpgrpc.WithDialOption(grpc.WithBlock()))
+		grpcOpts = append(grpcOpts, otlpgrpc.WithDialOption(grpc.WithBlock()))
```
Comment (tobert) [inline cmd/plumbing.go:59]: good rename, thank you

**Review by tobert — APPROVED** [review]
Looks good, thank you!!!

**Comment by ahayworth** [comment]
> There are unit tests in cmd/

Whoops 🤦

**Pull request merged**